File: backend/src/main/java/ma/dnaengineering/backend/model/Job.java
package ma.dnaengineering.backend.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "jobs")
@Data
public class Job {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull(message = "Title is required")
    @Column(nullable = false)
    private String title;

    @NotNull(message = "Description is required")
    @Column(columnDefinition = "TEXT", nullable = false)
    private String description;

    private String location;

    private BigDecimal salary;

    @Column(name = "posted_at")
    private LocalDateTime postedAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        postedAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

}

File: frontend/src/app/types/job.ts
export interface Job {
  id?: number;
  title: string;
  description: string;
  location?: string;
  salary?: number;
  postedAt?: string;
  updatedAt?: string;
}

export interface JobResponse {
  id: number;
  title: string;
  description: string;
  location: string;
  salary: number;
  postedAt: string;
  updatedAt: string;
}
